modules/market/src/main/java/com/opengamma/strata/market/curve/ParRates.java

```diff
@@ -57,6 +59,20 @@ public static ParRates of(Map<ObservableId, Double> rates, CurveMetadata metadat
     return new ParRates(rates, metadata);
   }
 
+  //-------------------------------------------------------------------------
+  /**
+   * Converts the par rates to a map by key.
+   *
+   * @return the equivalent map of par rate by key
+   */
+  public ImmutableMap<ObservableKey, Double> toRatesByKey() {
+    ImmutableMap.Builder<ObservableKey, Double> ratesByKey = ImmutableMap.builder();
```

Comment:
Could use:

    return rates.entrySet().stream()
        .map(e -> Pairs.of(e.getKey().toObservableKey(), e.getValue())
        .collect(pairsToImmutableMap());